fem/coefficient.hpp

```diff
@@ -946,6 +946,54 @@ class OuterProductCoefficient : public MatrixCoefficient
                      const IntegrationPoint &ip);
 };
 
+class QuadratureFunction;
+
+/** @brief Vector quadrature function coefficient which requires that the
+   quadrature rules used for this vector coefficient be the same as those that
+   live within the supplied QuadratureFunction. */
+class VectorQuadratureFunctionCoefficient : public VectorCoefficient
+{
+private:
+   const QuadratureFunction &QuadF; //do not own
+   int index;
+
+public:
+   /// Constructor with a quadrature function as input
+   VectorQuadratureFunctionCoefficient(QuadratureFunction &qf);
+
+   /** Set the starting index within the QuadFunc that'll be used to
+      project outwards as well as the corresponding length. The projected length
+      should have the bounds of 1 <= length <= (length QuadFunc - index). */
+   void SetComponent(int _index, int _length);
+
+   const QuadratureFunction *GetQuadFunction() const { return &QuadF; }
```

Comment:
Maybe return a reference as `QuadF` can't be NULL. Same on line 990.

Reply:
Updated it to a reference